### PR TITLE
Signup: add site preview in the onboarding flow

### PR DESCRIPTION
**Pull request by mattwiebe**
### Changes proposed in this Pull Request

Adds a `SiteMockup` component to the `onboarding-dev` flow for later inclusion in the `onboarding` flow. This will allow users to see a stylized mockup of how the information will be displayed in the Radcliffe 2 theme once they complete signup.

**Note** This is not yet hooked up to the verticals API endpoint, which will supply 

### Testing instructions

1. Enter the `onboarding-dev` flow via `/start/onboarding-dev/`
2. On a wide screen, both the desktop and mobile mockups should appear beneath the step content
3. On a small screen, only the mobile mockup should appear
4. In the `site-topic` step, choose "Mexican Restaurant" to move from outlines to seeing site mockup content
5. In the `site-information` step (after choosing **business** in the `site-type` step), you should see the name, address, and phone number reflected in the mockup
6. Try to break things by not following exactly the steps above.

Wide screen
image

Small screen
image

Site information step
image

### CODE REVIEW

**Comment by alisterscott** [comment]
The canary test failure is unrelated - it's expecting the new gutenberg routes from master

**Comment by adamziel** [comment]
> When you pick a vertical it's not clear that something is loading. Maybe we need to show some kind of progress indicator or loading state for images before they come in.

Here are some quick&dirty 5 minutes chrome inspector ideas:

loading
loading

> If you pick a vertical without content it shows the wireframe. We have default content for the business segment, a base site pack if you will, that we should probably show instead of the wireframes.

@fditrapani would you mind linking to that default content?

File: client/signup/main.jsx [hunk 1]
```diff
@@ -582,9 +581,8 @@ class Signup extends React.Component {
 							flowName={ this.props.flowName }
 						/>
 					) }
-				<TransitionGroup component="div" className="signup__steps">
```
Comment (adamziel) [inline client/signup/main.jsx]: Why are we getting rid of TransitionGroup here?

File: client/signup/main.jsx [hunk 2]
```diff
@@ -517,7 +516,7 @@ class Signup extends React.Component {
 		const shouldRenderLocaleSuggestions = 0 === this.getPositionInFlow() && ! this.props.isLoggedIn;
 
 		return (
-			<CSSTransition classNames="signup__step" timeout={ 400 } key={ stepKey }>
```
Comment (adamziel) [inline client/signup/main.jsx]: Why are we getting rid of CSSTransition here?

**Comment by fditrapani** [comment]
Thanks for those ideas @adamziel. Again, I think it's probably best to single those kinds of updates into their own PR so they're easier for review. 

Here's the base site content: https://a8cvm1.wordpress.com

**Comment by adamziel** [comment]
Agreed @fditrapani, I created a few separate trello tasks that we could view as extensions to the base version. Let's focus on ironing out the following items that sound more like things we need to iron out before merging the first version:

* When you first pick a vertical, the address and phone number show. Then when you add a title/phone/address, they disappear. It might be better to not show the tagline content until people add their own details in.
* The preview is showing for the Blog, Pro, Educator segments.
* I don't know if this is a regression but if you pick online store, you can't go back to the segmentation step again.

(btw I wasn't able to reproduce the last one - back button works fine for me both in this PR and in production)

File: client/signup/steps/site-topic/index.jsx [hunk 3]
```diff
@@ -60,6 +60,9 @@ class SiteTopicStep extends Component {
 
 	onSiteTopicChange = value => {
 		this.setState( { siteTopicValue: value } );
+		if ( this.props.flowName === 'onboarding-dev' ) {
```
Comment (scruffian) [inline client/signup/steps/site-topic/index.jsx]: I assume these changes are unrelated?

File: client/signup/steps/site-information/index.jsx [hunk 4]
```diff
@@ -53,6 +53,11 @@ class SiteInformation extends Component {
 
 	handleInputChange = ( { target: { name, value } } ) => {
 		this.setState( { [ name ]: value } );
+		if ( this.props.flowName === 'onboarding-dev' ) {
```
Comment (scruffian) [inline client/signup/steps/site-information/index.jsx]: I think it's better to create a specific version of this step for use in the onboarding flow, rather than having `this.props.flowName === 'onboarding-dev'` in the code. You can create new steps by reusing existing components.

**Comment by shaunandrews** [comment]
I think the loading state only needs to be applied to the images — it seems the copy won't need it. Will there likely be a delay in loading content, @mattwiebe?

--

When searching on the `site-type` step I noticed that the list of suggestions affects the flow of the page, causing the screen to scroll and push the SiteMockup down the page.

screen shot 2018-12-05 at 10 36 54 am

Can we pull it out of the flow of the page, and position it to the input? Then we can give it a max-height and it let it scroll independently of the page.

--

I think we hide the SiteMockup component until you type something into the `site-type` input. Then we can bring back the fun transition to introduce the SiteMockup.

preview intro

File: client/signup/site-mockup/site-mockup.jsx [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+
+/**
+ * Internal dependencies
+ */
+import { translate } from 'i18n-calypso';
+
+/**
+ * Style dependencies
+ */
+import './site-mockup.scss';
```
Comment (niranjan-uma-shankar) [inline client/signup/site-mockup/site-mockup.jsx]: Should the contents of this file rather be included in `site-mockup/style.scss`? There was an effort to move all scss files to webpack in #27515, and the related work in signup was done in #28394, so that could require a naming convention to be followed?

**Comment by adamziel** [comment]
@shaunandrews just giving you heads up that we have a separate card for the "preview being pushed down" thing

File: client/signup/main.jsx (hunk 1 above)
Comment (mattwiebe) [reply client/signup/main.jsx]: Hey @shaunandrews - what was the reasoning on this? 

I think it's because `TransitionGroup` does some absolute positioning so that we can't just have the previews underneath. That would, I assume apply to the `CSSTransition` below, too.

File: client/signup/steps/site-information/index.jsx (hunk 4 above)
Comment (mattwiebe) [reply client/signup/steps/site-information/index.jsx]: I should have left a code comment to indicate that this was just temporary. The `onboarding` flow is in live testing already and I don't want this to appear before it's fully baked. This is just to start getting things landed and testable in a flow we're not sending any traffic to.

File: client/signup/site-mockup/site-mockup.jsx (hunk 5 above)
Comment (mattwiebe) [reply client/signup/site-mockup/site-mockup.jsx]: The notion here is just that the `SiteMockup` component should be considered separately from the way it is being implemented within the signup flow itself. This means that it probably just be moved into `client/components/site-mockup`

File: client/signup/steps/site-information/index.jsx (hunk 4 above)
Comment (scruffian) [reply client/signup/steps/site-information/index.jsx]: 👍

Comment (mattwiebe) [reply client/signup/steps/site-information/index.jsx]: Oh and in this particular case I wanted to better profile if this needed debouncing/throttling and I just haven't gotten to it yet.

File: client/signup/main.jsx (hunk 1 above)
Comment (shaunandrews) [reply client/signup/main.jsx]: Yes, its related to the positioning of the previews. There's likely some workarounds we could do to keep the transitions (as they exist in production) in place, but it felt like too much work for little benefit at this time.

Once we have a fixed-height area for each step, then it will be easier to bring the transitions back.

**Comment by mattwiebe** [comment]
Just focusing on the stuff that @adamziel brought up above that needs addressing in this PR:

> When you first pick a vertical, the address and phone number show. Then when you add a title/phone/address, they disappear. It might be better to not show the tagline content until people add their own details in.

Not sure what this means, @fditrapani ? What happens right now with a clean browser session is that you see *You’ll be able to customize this to your needs* in the tagline area until the address and/or phone # is entered. I did find a bug where it would then be empty in the case where the title is modified but the address/phone not yet entered, which should be fixed now.

If we decide we don't want a default tagline (or to modify it), happy to do that, too.

> The preview is showing for the Blog, Pro, Educator segments.

Ah, hadn't noticed we wanted to restrict to business. Done.

> I don't know if this is a regression but if you pick online store, you can't go back to the segmentation step again.

Not a regression, that's happening in the `onboarding` flow. too. It's what comes into play when we redirect from one flow into the beginning of another. As far as the `eccomerce` flow is concerned, there is nothing to go back to, which is why it doesn't show the `← Back` link (browser back will still work fine). @taggon did some similar sleuthing in and around a similar issue a while back with the `store-nux` flow in #24912

**Comment by fditrapani** [comment]
> Not sure what this means, @fditrapani ? What happens right now with a clean browser session is that you see You’ll be able to customize this to your needs in the tagline area until the address and/or phone # is entered.

Looks like you made an update here and the there's copy under the title that says "You’ll be able to customize this to your needs." — that works! What I noticed this time though is while the title changes when you add the business name, the rest of the copy doesn't update when you add an address or phone number. Is this intentional?  

image



> Not a regression, that's happening in the onboarding flow. too.

Cool, got it. Something for us to address at another time.

**Comment by mattwiebe** [comment]
> What I noticed this time though is while the title changes when you add the business name, the rest of the copy doesn't update when you add an address or phone number. Is this intentional?

Nope, that was a regression. I think I must have tested a version that I hadn't reloaded. Yay compiling! 
f3cd909  fixes it and it should be working as expected.

File: client/signup/steps/site-topic/index.jsx (hunk 3 above)
Comment (michaeldcain) [reply client/signup/steps/site-topic/index.jsx]: I think he's doing this to test updating the topic (and the preview) onChange, not just onSubmit.
